src/sinks/util/buffer/metrics.rs

```diff
@@ -108,83 +110,84 @@ impl Batch for MetricBuffer {
     type Input = Event;
     type Output = Vec<Metric>;
 
-    fn len(&self) -> usize {
-        self.num_items()
-    }
-
-    fn push(&mut self, item: Self::Input) {
-        let item = item.into_metric();
-
-        match &item.value {
-            MetricValue::Counter { value } if item.kind.is_absolute() => {
-                let new = MetricEntry(item.clone());
-                if let Some(MetricEntry(Metric {
-                    value: MetricValue::Counter { value: value0, .. },
-                    ..
-                })) = self.state.get(&new)
-                {
-                    // Counters are disaggregated. We take the previoud value from the state
-                    // and emit the difference between previous and current as a Counter
-                    let delta = MetricEntry(Metric {
-                        name: item.name.to_string(),
-                        timestamp: item.timestamp,
-                        tags: item.tags.clone(),
-                        kind: MetricKind::Incremental,
-                        value: MetricValue::Counter {
-                            value: value - value0,
-                        },
-                    });
-
-                    // The resulting Counters could be added up normally
-                    if let Some(MetricEntry(mut existing)) = self.metrics.take(&delta) {
+    fn push(&mut self, item: Self::Input) -> PushResult<Self::Input> {
+        if self.num_items() >= self.max_events {
+            PushResult::Full(item)
```

Comment:
True. I could rework the logic to push that decision into each of the metric types below if you think it is worth it. It would likely be a pretty small batch size optimization though.

Reply:
I don't think it's worthwhile right now. Maybe a later optimization.